### PR TITLE
[Release 1.19] Adding configuration to run kube proxy as a static pod.

### PR DESCRIPTION
**Pull request by luthermonson**
#### Linked Issues ####
https://github.com/rancher/rke2/issues/1198

### CODE REVIEW

File: pkg/rke2/serviceaccount.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+	"time"
+
+	"github.com/sirupsen/logrus"
+	v1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/kubernetes/plugin/pkg/admission/serviceaccount"
```
Comment (manuelbuil) [inline pkg/rke2/serviceaccount.go:15]: It seems this library is missing in go.mod

File: pkg/podexecutor/staticpod.go [hunk 2]
```diff
@@ -81,9 +81,23 @@ func (s *StaticPodConfig) Kubelet(args []string) error {
 	return nil
 }
 
-// KubeProxy panics if used. KubeProxy for RKE2 is provided by a packaged component (rke2-kube-proxy Helm chart).
+// KubeProxy starts Kube Proxy as a static pod.
 func (s *StaticPodConfig) KubeProxy(args []string) error {
-	panic("kube-proxy unsupported")
+	image, err := s.Resolver.GetReference(images.KubeProxy)
```
Comment (manuelbuil) [inline pkg/podexecutor/staticpod.go:86]: images.KubeProxy is undefined. I'm not familiar with the code but StaticPodConfig struct contains the field `Images`, maybe that's the one you wanted to refer to?

File: pkg/podexecutor/staticpod.go [hunk 3]
```diff
@@ -81,9 +81,23 @@ func (s *StaticPodConfig) Kubelet(args []string) error {
...
+	if err != nil {
+		return err
+	}
+	if err := images.Pull(s.ImagesDir, images.KubeProxy, image); err != nil {
```
Comment (manuelbuil) [inline pkg/podexecutor/staticpod.go:90]: Same as above, `images.KubeProxy` is undefined

**Review by manuelbuil — CHANGES_REQUESTED** [review]
I ran `go test -v .` on the code and found these issues

**Comment by luthermonson** [comment]
we decided that we are not backporting the kube-proxy staticpod to 1.19, too many changes with image resolver etc. that we're not going to bother as it is EOL is very soon

**Pull request closed without merging**